tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -203,7 +203,7 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
   /** Desugars parameters and body to a JS function.
    */
   private[emitter] def desugarToFunction(
-      classEmitter: ScalaJSClassEmitter, enclosingClassName: String,
+      classEmitter: ScalaJSClassEmitter, enclosingClassName: Option[String],
```

Comment:
That's right. I'll try to extract that in a separate commit. It shouldn't be too hard.